### PR TITLE
MM-68154: Upgrade imagemeta to v0.17.2

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary

- Upgrades `github.com/bep/imagemeta` from `v0.12.0` to `v0.17.2`
- Replaces the forward-only `fwSeeker` with `bufReadSeeker`, which buffers reads to support backward seeks required by imagemeta v0.17.x's EXIF offset-based tag decoding
- Switches to `imagemeta.ErrStopWalking` (library-provided sentinel) instead of a custom stop error; updates `Decode` call to capture its new `(DecodeResult, error)` return signature

#### Ticket Link

Fixes: https://mattermost.atlassian.net/browse/MM-68154

#### Release Note

```release-note
Upgraded github.com/bep/imagemeta dependency from v0.12.0 to v0.17.2.
```

## Change Impact: 🟡 Medium

**Regression Risk:** Upgrading imagemeta and replacing the forward-only seeker with a buffered ReadSeeker changes EXIF parsing behavior for non-seekable inputs and EXIF offset decoding; tests cover the new seeker and common cases but edge-case image formats or unusual stream behaviors could surface regressions. The dependency API change and altered error handling introduce modest risk where callers previously relied on prior decode semantics.

**QA Recommendation:** Moderate manual QA: exercise image upload/display flows with varied JPEG/PNG/TIFF/WebP samples (large/malformed EXIF, streamed/non-seekable inputs) and run the updated unit tests to validate orientation detection and error behaviors.

*Generated by CodeRabbitAI*

---

**Release Note:** Upgraded github.com/bep/imagemeta dependency from v0.12.0 to v0.17.2.

[Review Change Stack](https://app.coderabbit.ai/change-stack/mattermost/mattermost/pull/36588?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: server/channels/app/imaging/orientation.go [hunk 1]
```diff
@@ -59,49 +57,77 @@ func MakeImageUpright(img image.Image, orientation int) image.Image {
...
-	f.pos += n
-
-	return f.pos, nil
+	if newPos > maxExifScanSize {
+		return b.pos, fmt.Errorf("seek: target %d exceeds %d-byte scan limit", newPos, maxExifScanSize)
+	}
+	// Seek forward past buffered data: read ahead into the buffer in-place.
+	toRead := int(newPos - int64(len(b.buf)))
+	oldLen := len(b.buf)
+	b.buf = append(b.buf, make([]byte, toRead)...)
```
Comment (carlisgg) [inline server/channels/app/imaging/orientation.go]: instead of growing the slice via `append+make` we could use `slices.Grow`

File: server/channels/app/imaging/orientation.go [hunk 2]
```diff
@@ -59,49 +57,77 @@ func MakeImageUpright(img image.Image, orientation int) image.Image {
...
 	r   io.Reader
+	buf []byte
 	pos int64
 }
 
-func (f *fwSeeker) Read(p []byte) (int, error) {
-	n, err := f.r.Read(p)
-	if err != nil {
-		return n, err
+// maxExifScanSize mirrors imagemeta's internal maxBufSize (10 MB); imagemeta
```
Comment (carlisgg) [inline server/channels/app/imaging/orientation.go]: maybe this comment is misleading. ´imagedata´ imposes that limit when reading a particular chunk the library wants to buffer and not a max read from the beginning of the file. The max scan size here still prevent retaining bytes that would compromise memory while scanning so it is great but maybe the comment should be more generic like  `maxExifScanSize prevents buffer growing over a reasonable limit since EXIF lives near the file start in all common formats` or similar

**Review by carlisgg — COMMENTED** [review]
it looks goot, just 2 small issues I have thought about while rereading the code

File: server/channels/app/imaging/orientation.go (hunk 1 above)
Comment (lieut-data) [reply server/channels/app/imaging/orientation.go]: Nice, thanks!

**Comment by lieut-data** [comment]
@DHaussermann, based on your last round of testing around profile pictures, I'm inclined to think our existing test coverage suffices, but I wonder if you could perform your risk analysis on this PR to decide how much additional review is required?

**Review by carlisgg — APPROVED** [review]
LGTM

**Comment by lieut-data** [comment]
Oops, apologies @DHaussermann -- auto-merge caught me again. Would appreciate your eyes on the risk analysis, but we'll close this one in `master` on Community as well.

**Comment by DHaussermann** [comment]
Thanks for the ping @lieut-data. I can have a look and merge test in `master`. I'll follow-up if there are any issues.
